src/uct/ib/rdmacm/rdmacm_cm.c

```diff
@@ -201,7 +211,8 @@ static ucs_status_t uct_rdmacm_cm_id_to_dev_addr(struct rdma_cm_id *cm_id,
                                 UCT_IB_ADDRESS_PACK_FLAG_SUBNET_PREFIX;
     }
 
-    params.gid  = cm_id->route.addr.addr.ibaddr.dgid;
+    params.gid  = qp_attr.ah_attr.is_global ? qp_attr.ah_attr.grh.dgid :
+                  cm_id->route.addr.addr.ibaddr.dgid;
```

Comment:
why need to check is_global?

Reply:
else if (is_global)  {
  set gid for qpattr
  set pack flags UCT_IB_ADDRESS_PACK_FLAG_INTERFACE_ID..
}